### DISCUSSION TITLE
Add custom styling to the video controls

### DISCUSSION THREAD

**Discussion by Alexandria** [TheWidlarzGroup/react-native-video#4949] [Ideas]
### What problem are you trying to solve?

Would love to be able to change the styling of the controls to better match the theme or branding of an app.

<img width="254" alt="17-05-m0mgj-hprvd" src="https://user-images.githubusercontent.com/7613670/185243053-1edf145d-548a-4ad3-bd6f-62c0ae11bf36.png">

For example I would want to lower the opacity of the controls so that just the speaker icon is visible. Or change the color of the play bar etc.

### Proposed solution

Add the ability to override the default styles of the video controls (on both iOS and Android).

Just at a quick glance of the code I saw styles for `seekbarFill` and `seekbarHandle` in the `VideoPlayer.andriod.tsx` file. Would it be possible to have the option to override the default styles here? (as well as with ios)

#### Code sample

Ideally it could look something like this when implemented

```javascript
// Load the module
import Video from 'react-native-video';
<Video source={{uri: "background"}}   
       ref={(ref) => {
         this.player = ref
       }}                                      
       onBuffer={this.onBuffer}              
       onError={this.videoError}               
       style={styles.backgroundVideo} />
       controls
       styleControls={
              seekbarTrackColor: "green",   // This would override seekbarTrack.backgroundColor: '#333',
              seekbarTrackWidth: "100%"    // This would override seekbarTrack.Width
              seekbarHandleColor: "black"  // This would override seekbarHandle.backgroundColor: '#333',
              topControls:"row"  // This could also be column left aligned or column right aligned. 
       }
/>
// Later on in your styles..
var styles = StyleSheet.create({
  backgroundVideo: {
   ......
  },
});
```

### Platforms

iOS, Android

**Comment by Alexandria** [comment]
Im happy to submit a PR and work on this.

**Comment by freeboub** [comment]
> 

I would be happy to merge it ! :) 
Thank you !

**Comment by nikitph** [comment]
@Alexandria is this something in progress on your end? Would love to see this

**Comment by codefromahmad** [comment]
@Alexandria I am also waiting for it.

**Comment by mitchiemt11** [comment]
@Alexandria was this merged?

**Comment by mitchiemt11** [comment]
@codefromahmad did you get this to work? I'm also still waiting for this.

**Comment by AhmedMustafa505** [comment]
Is this working now ???

**Comment by TriptiPant01** [comment]
Any Progress on this?

**Comment by aman-technyx** [comment]
still looking this feature !!

**Comment by amitojsingh** [comment]
any update on this?

**Comment by saintyoseph** [comment]
```
controlsStyles={{
              hideFullscreen: true,
              hidePlayPause: true,
            }}
```

These are not working...

**Comment by sayan-avantai** [comment]
in **controlsStyles** almost nothing is working.
          hideNext: true,
          hidePrevious: true,
          hideSettingButton: true, etc.

package : 

```
"react": "18.3.1",
"react-native": "0.76.2",
"react-native-video": "^6.17.0",
```

**Comment by davitHarut** [comment]
It doesn't work for me either
```
controlsStyles={{
      hideFullscreen: true,
      hidePlayPause: true,
      hideDuration: true,
}}
```

```
"react": "19.1.0",
 "react-native": "0.81.1",
  "react-native-video": "^6.17.0",
```

**Comment by kartvya** [comment]
Not for me either 

controlsStyles={{
              hideForward:true,
              hideNext:true,
              hideSeekBar:true,
              hideSettingButton:true
            }}

not working 

"react-native": "0.82.1",
"react-native-video": "^6.17.0",

**Comment by TWG-AI** [comment]
Thanks for the idea — noted for the team to consider.

> I'm an AI assistant, so double-check any specifics I may have mentioned.
